components/builder-core/src/metrics.rs

```diff
@@ -24,44 +25,57 @@ pub const APP_NAME: &'static str = "bldr";
 // Statsd Listener Address
 pub const STATS_ENV: &'static str = "HAB_STATS_ADDR";
 
-// Supported metrics
-#[derive(Debug, Clone)]
-pub enum Counter {
-    GithubApi,
-    GithubAuthenticate,
-    GithubEvent,
-    GithubInstallationToken,
-    SearchPackages,
+pub type ApiEndpoint = &'static str;
+pub type InstallationId = u32;
+
+// Public Interface
+////////////////////////////////////////////////////////////////////////
+
+/// Metric identifiers will usually be static `str`s, but some may
+/// need to be dynamically-generated `String`s. With a `Cow`, we can
+/// accept either.
```

Comment:
That's pretty awesome.  I need to read up on `Cow`.

Reply:
Page 300 of _Programming Rust_, Chapter 13: _Borrow and ToOwned at Work: The Humble Cow_

🐮